lib/KIcon/iconDefinitions.js

```diff
@@ -0,0 +1,268 @@
+  insert_chart: { icon: require('./material-svg/insert_chart/Baseline.vue').default },
+  // DeviceTopNav
+  https: { icon: require('./material-svg/https/Baseline.vue').default },
+  perm_device_information: {
+    icon: require('./material-svg/perm_device_information/Baseline.vue').default,
+  },
+  settings: { icon: require('./material-svg/settings/Baseline.vue').default },
+  // SettingsButton
+  tune: { icon: require('./material-svg/tune/Baseline.vue').default },
+  // TocButton
+  list: { icon: require('./material-svg/list/Baseline.vue').default },
+  // TopBar (and likely others?)
+  fullscreen: { icon: require('./material-svg/fullscreen/Baseline.vue').default },
+  fullscreen_exit: { icon: require('./material-svg/fullscreen_exit/Baseline.vue').default },
+  //SyncInterface
+  verified_user: { icon: require('./material-svg/verified_user/Baseline.vue').default },
+  //FacilityTopNav
+  save: { icon: require('./material-svg/save/Baseline.vue').default },
+  domain: { icon: require('./material-svg/domain/Baseline.vue').default },
+  // LearnTopNav
+  forum: { icon: require('./material-svg/forum/Baseline.vue').default },
+  // Searchbox
+  filter_list: { icon: require('./material-svg/filter_list/Baseline.vue').default },
+  warning: { icon: require('./material-svg/warning/Baseline.vue').default },
+};
+
+/**
+ * Given one of `blackPaths` or `coloredPaths` from `allIcons`,
+ * returns array where each element is an array of aliases which
+ * refer to the same icon. The sub-arrays are already sorted
+ * alphabetically.
+ *
+ * Example:
+ *
+ * [
+ *     ['arrow_back', 'back'],
+ *     ['arrow_forward', 'forward'],
+ *     ...
+ *     ['verified_user']
+ * ]
+ *
+ */
+
+const sortedIcons = paths => {
+  return Object.keys(paths)
+    .map(a => paths[a])
+    .sort((a, b) => a[0].charCodeAt(0) - b[0].charCodeAt(0));
+};
+
+/*
+ * Assigned the value: `{ blackPaths, coloredPaths }` from the function below.
+ *
+ * `blackPaths` and `coloredPaths` are objects wherein the keys are unique `paths` to an icon
+ *
+ * These path keys map to an array of alphabetically sorted aliases associated with that path.
+ */
+
+const allIcons = (() => {
+  let blackPaths = {};
+  let coloredPaths = {};
+
+  Object.keys(KolibriIcons).forEach(alias => {
+    const path = KolibriIcons[alias].icon.__file;
+    const hasColorToken = KolibriIcons[alias].defaultColor;
+    const thePathsWeCareAbout = hasColorToken ? coloredPaths : blackPaths;
+
+    let aliases = [alias];
+
+    if (Object.keys(thePathsWeCareAbout).includes(path)) {
+      aliases = [...thePathsWeCareAbout[path], ...aliases].sort();
+    }
+
+    thePathsWeCareAbout[path] = aliases;
+
+    if (hasColorToken) {
+      coloredPaths = thePathsWeCareAbout;
+    } else {
+      blackPaths = thePathsWeCareAbout;
+    }
+  });
+
+  return { blackPaths, coloredPaths };
+})();
+
+const sortedColoredIcons = sortedIcons(allIcons.coloredPaths);
+const sortedBlackIcons = sortedIcons(allIcons.blackPaths);
```

Comment:
I'm a little confused why we're exporting the file paths of the components, e.g. `KolibriIcons[alias].icon.__file`. Where would this be used?

Reply:
`allIcons.coloredPaths` and `allIcons.blackPaths` use that file path as a unique identifier. All aliases that point to the same `KolibriIcons[alias].icon.__file` are aliases for the same icon. Then there's some accounting for whether it has a `defaultColor` defined on it.

```
{
    '/path/to/a/specific/icon_type/Baseline.vue': [ 'alias1', 'alias2' ],
    '/path/to/a/different/icon_type/Outline.vue': ['differentAlias'],
    ...
}
```

We're exporting the return value of passing those objects to `sortedIcons()` which turns it into a sorted array of sorted arrays:

```
[
    [ 'channel'],
    [ 'coach', 'coachContents'],
    ...
    ['zebra']
]
```

Ultimately, this is all just so that you can `v-for` that outermost array where the iterator is an array of aliases which all refer to the same KIcon icon.

```
<div v-for="aliasArray in sortedBlackIcons">
    <KIcon :icon="aliasArray[0]" />
    <span> {{ aliasArray.join(", ") }} </span>
</div>
```

Reply:
FWIW here is what KIconDebug looks like:

![image](https://user-images.githubusercontent.com/6356129/85618835-3ac8df00-b616-11ea-8abd-56a21aaecbcb.png)